include/numerics/dense_matrix.h

```diff
@@ -31,7 +31,11 @@
 # ifdef I
 #  define LIBMESH_SAW_I
 # endif
+
+#include "libmesh/ignore_warnings.h"
 # include <petscsys.h>
```

Comment:
What's getting triggered by their header that wasn't before?

Reply:
The warning was:
```
    In file included from ./include/libmesh/petsc_solver_exception.h:29,
                     from ./include/libmesh/petsc_unique_ptr.h:26,
                     from ../src/partitioning/partitioner.C:29:
    /opt/moose/petsc-3.11.4/mpich-3.3_gcc-9.2.0-opt/include/petscsys.h: In function 'PetscErrorCode PetscMemcpy(void*, const void*, std::size_t)':
    /opt/moose/petsc-3.11.4/mpich-3.3_gcc-9.2.0-opt/include/petscsys.h:1650:32: error: cast from type 'const void*' to type 'char*' casts away qualifiers [-Werror=cast-qual]
     1650 |     memcpy((char*)(a),(char*)(b),n);
          |                                ^
    In file included from ./include/libmesh/petsc_solver_exception.h:29,
                     from ./include/libmesh/petsc_unique_ptr.h:26,
                     from ../src/partitioning/partitioner.C:29:
    /opt/moose/petsc-3.11.4/mpich-3.3_gcc-9.2.0-opt/include/petscsys.h: In function 'PetscErrorCode PetscSegBufferGetInts(PetscSegBuffer, PetscInt, PetscInt* __restrict__*)':
    /opt/moose/petsc-3.11.4/mpich-3.3_gcc-9.2.0-opt/include/petscsys.h:2389:166: error: cast from type 'PetscInt* __restrict__*' {aka 'int* __restrict__*'} to type 'void**' casts away qualifiers [-Werror=cast-qual]
     2389 | PETSC_STATIC_INLINE PetscErrorCode PetscSegBufferGetInts(PetscSegBuffer seg,PetscInt count,PetscInt *PETSC_RESTRICT *slot) {return PetscSegBufferGet(seg,(size_t)count,(void**)slot);}
          |                                                                                                                                                                      ^~~~
    cc1plus: all warnings being treated as errors
```
I actually only saw it on the new header file I added, not quite sure why it wouldn't be triggered from the other inclusions of the same header, but I felt it was better to be on the safe side. It also doesn't seem to be triggered in newer PETScs.